pytorch_lightning/plugins/precision/native_amp.py

```diff
@@ -106,10 +107,23 @@ def forward_context(self) -> Generator[None, None, None]:
         with self.autocast_context_manager():
             yield
 
+    def state_dict(self) -> Dict[str, Any]:
+        if self.scaler is not None:
+            return self.scaler.state_dict()
+        return {}
+
+    def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
+        if self.scaler is not None:
+            self.scaler.load_state_dict(state_dict)
+
     def on_load_checkpoint(self, checkpoint: Dict[str, Any]) -> None:
-        if self.scaler is not None and "native_amp_scaling_state" in checkpoint:
-            self.scaler.load_state_dict(checkpoint["native_amp_scaling_state"])
+        rank_zero_deprecation(
```

Comment:
Missing deprecation tests for these 2 here and in the apex plugin

Reply:
so these are  hooks called by Lightning, not the user. Only if the user overrides these hooks AND calls super will they every see this warning, which is less likely. 

I would put the warning into the docstring instead. 
Optionally one can use the is_overridden utility in the place where this hook gets called.

Reply:
- protecting against `super.on_save/load_checkpoint()` calls: presumably this would only happen inside an overriden `on_save/load_checkpoint`. 
- protecting against overriding `on_save/load_checkpoint()`: this itself actually does not need any warnings - until issue #11824

we can either do
1. `if is_overridden: deprecation("if you are calling super.on_save/load_checkpoint, super method is deprecated.)`, then after issue #11824 the warning message just becomes `"checkpointhooks are deprecated"`
2. or we can just do docstring for now until  `"checkpointhooks are deprecated"` entirely

I'm thinking 2

Reply:
I think docstring for now and deprecation in a follow up is reasonable

Reply:
yes